server/asset/eth/rpcclient.go

```diff
@@ -8,28 +8,37 @@ import (
 	"fmt"
 	"math/big"
 
+	"github.com/ethereum/go-ethereum"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/p2p"
 	"github.com/ethereum/go-ethereum/rpc"
 )
 
 // Check that rpcclient satisfies the ethFetcher interface.
 var _ ethFetcher = (*rpcclient)(nil)
 
 type rpcclient struct {
+	// ec wraps the client with some useful calls.
 	ec *ethclient.Client
+	// c is a direct client for raw calls.
+	c *rpc.Client
 }
 
+const gweiFactor = 1e9
+
+var gweiFactorBig = big.NewInt(gweiFactor)
```

Comment:
These aren't really specific to the RPC client. Can you move these to **eth.go**. 

My guess would be that the conversion logic you have in `FeeRate` will end up as a function something like

```Go
func toGwei(wei *big.Int) (uint64, error) {
	bigGP.Div(wei, gweiFactorBig)
	if !bigGP.IsUint64() {
		return 0, fmt.Errorf("suggest gas price %v gwei is too big for a uint64", bigGP)
	}
	return bigGP.Uint64(), nil
}
```

which will be used mostly in **eth.go**.

Reply:
moving